src/redux/slice.ts

```diff
@@ -0,0 +1,54 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { CoinQuantity } from "fuels";
+import { CoinWithInfo } from "@/hooks/useCoins";
+import { Stream } from "@/hooks/Streams";
+
+// Define the type for your slice state
+type PipelineState = {
+  coins: CoinQuantity[];
+  coinsWithInfo: CoinWithInfo[];
+  globalStreams: Stream[];
+  sendingOrReceiving: string;
+  timezone: string | null;
+};
+
+// Set the initial state with the defined type
+const initialState: PipelineState = {
+  coins: [] as CoinQuantity[],
+  coinsWithInfo: [] as CoinWithInfo[],
+  globalStreams: [] as Stream[],
+  sendingOrReceiving: "",
+  timezone: null,
+};
+
+export const slice = createSlice({
+  name: "pipeline",
+  initialState,
+  reducers: {
```

Comment:
Hmm this doesn't make sense to me as the implementation.

Coins should be a slice,
sendingOrReceiving should be a slice,
Streams should be a slice
Timezone should be a slice